### DISCUSSION TITLE
Best approach to service_role policy for private bucket

### DISCUSSION THREAD

**Discussion by yafimski** [supabase/supabase#23840] [Questions] [answered]
Hi,
I'm trying to understand if this is a decent policy for a **private** bucket that limits all actions just for **service_role** that uses the **SERVICE_KEY**:

```
create policy "Service Role has full access to bucket 'bucketA'"
on storage.objects
for select, insert, update, delete
to service_role
using (
  bucket_id = 'bucketA'
);
```

What do you think? I understand from reading docs/info that if the bucket is private, it is by default using RLS..?
Do I even need this policy or can I have the bucket without policies since it's private..?

Thanks

**Comment by GaryAustin1** [comment]
If you are using service role you do not need policies.  No policy prevents access to anon and authenticated.  Service_role bypasses RLS so does not care what you put there.  Note to service_role is a waste, it does nothing.  Service_role does not even look at the policy at all.

**Reply by yafimski** [reply]
Thanks @GaryAustin1 

> No policy prevents access to anon and authenticated.

From what you say I understand that if anon+authenticated are used, the bucket is totally exposed...? 
Not 100% clear on if I got you correctly.. what you think might be the best way to restrict access completely to anything but calls from service_role.

**Reply by GaryAustin1** [reply, marked as answer]
The opposite.  Unless you add a policy TO public,anon or authenticated, they will have NO access.  So don't add a policy at all.
